src/components/UpgradePlanForm/UpgradePlanForm.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import axios from 'axios';
+import { FormattedMessage } from 'react-intl';
+
+class UpgradePlanForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      userPlanModel: null,
+      formIsValid: false,
+    };
+
+    this.submitForm = this.submitForm.bind(this);
+  }
+
+  componentWillMount() {
+    const idUserType = this.props.isSubscriber ? 4 : 2;
+    axios
+      .get(
+        process.env.REACT_APP_API_URL +
+          '/SendUpgradePlanContactEmail/GetUpgradePlanData?idUserType=' +
+          idUserType,
+        {
+          withCredentials: true,
+        },
+      )
+      .then((response) => {
+        this.setState({ userPlanModel: response.data.data });
+      });
+  }
+
+  submitForm() {
+    fetch(process.env.REACT_APP_API_URL + '/SendUpgradePlanContactEmail/SendEmailUpgradePlan', {
+      method: 'post',
+      crossDomain: true,
+      body: JSON.stringify(this.state.userPlanModel),
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      credentials: 'include',
+    })
+      .then((res) => {
+        return res.json();
+      })
+      .then((data) => {
+        this.props.handleClose();
```

Comment:
Why two _then_ in place of only one? the result of the first one will be lost, or am I missing something?

If you refactor it using _async/await_ it will be easy to understand.

Reply:
i change axios in post method for fetch because cancel request it's a axios problem and can't resolve.I use directly fetch from react

Reply:
You are using ES6 _fetch_, is it compatible with all supported browsers?

Sadly it is not so _test friendly? as Axios, but we could deal with it later.

I understood _credentials vs includeCredentials_, what about the two _then_?

Reply:
i don't know when use includeCredentials.. maybe it's old?? in documentation use credentials 
https://developer.mozilla.org/es/docs/Web/API/Fetch_API/Utilizando_Fetch

Reply:
![image](https://user-images.githubusercontent.com/8861133/53902214-7e619900-401f-11e9-97cf-0bc269371375.png)

support edge and not internet explorer